dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -1218,4 +1239,118 @@ public void updateUserReferences(String userId, String userRoleId, String replac
+		Role anyWhoCanViewContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_VIEW_ROLE_KEY);
+		Role anyWhoCanEditContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_EDIT_ROLE_KEY);
+		Role anyWhoCanPublishContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_PUBLISH_ROLE_KEY);
+		Role anyWhoCanEditPermisionsContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_EDIT_PERMISSIONS_ROLE_KEY);
+
+		if ((user != null) && roleAPI.doesUserHaveRole(user, roleAPI.loadCMSAdminRole()))
+			return actions;
+
+		List<WorkflowAction> permissionables = new ArrayList<>(actions);
+		if(permissionables.isEmpty()){
+			return permissionables;
+		}
+
+		WorkflowAction action;
+		int i = 0;
+
+		while (i < permissionables.size()) {
+			action = permissionables.get(i);
+			boolean havePermission = false;
+			if(null != permissionable) {
+				/* Validate if the action has one of the workflow special roles*/
+                havePermission = hasSpecialWorkflowPermission(user, respectFrontEndRoles, permissionable,
+                        anyWhoCanViewContent,
+                        anyWhoCanEditContent, anyWhoCanPublishContent,
+                        anyWhoCanEditPermisionsContent, action);
+
+			}
+			/* Validate if has other rolers permissions */
+			if(APILocator.getPermissionAPI().doesUserHavePermission(action, PermissionAPI.PERMISSION_USE, user, respectFrontEndRoles)){
+				havePermission=true;
+			}
+
+			/* Remove the action if the user dont have permission */
+			if(!havePermission){
+				permissionables.remove(i);
+			}else {
+				++i;
+			}
+		}
+
+		return permissionables;
+	}
+
+    /**
+     * Return true if the action has one of the workflow action roles and if the user has  any of
+     * those permission over the content or content type
+     * @param user User to validate
+     * @param respectFrontEndRoles indicates if should respect frontend roles
+     * @param permissionable ContentType or contentlet to validate special workflow roles
+     * @param anyWhoCanViewContent Workflow action role
+     * @param anyWhoCanEditContent Workflow action role
+     * @param anyWhoCanPublishContent Workflow action role
+     * @param anyWhoCanEditPermisionsContent Workflow action role
+     * @param action The action to validate
+     * @return true if the user has one of the special workflow action role permissions, false if not
+     * @throws DotDataException
+     */
+    @CloseDBIfOpened
+    private boolean hasSpecialWorkflowPermission(User user, boolean respectFrontEndRoles,
+            Permissionable permissionable, Role anyWhoCanViewContent, Role anyWhoCanEditContent,
+            Role anyWhoCanPublishContent, Role anyWhoCanEditPermisionsContent,
+            WorkflowAction action) throws DotDataException {
+
+    	if (APILocator.getPermissionAPI()
+                .doesRoleHavePermission(action, PermissionAPI.PERMISSION_USE,
+                        anyWhoCanViewContent)) {
+            if (APILocator.getPermissionAPI()

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)